src/Database/LSMTree/Internal/Serialise/Class.hs

```diff
@@ -87,10 +105,50 @@ class SerialiseValue v where
 serialiseValueIdentity :: (Eq v, SerialiseValue v) => v -> Bool
 serialiseValueIdentity x = deserialiseValue (serialiseValue x) == x
 
+-- | Test the __Identity up to slicing__ law for the 'SerialiseValue' class
+serialiseValueIdentityUpToSlicing ::
+     (Eq v, SerialiseValue v)
+  => RawBytes -> v -> RawBytes -> Bool
+serialiseValueIdentityUpToSlicing prefix x suffix =
+    deserialiseValue (packSlice prefix (serialiseValue x) suffix) == x
+
 -- | Test the __Concat distributes__ law for the 'SerialiseValue' class
 serialiseValueConcatDistributes :: forall v. (Eq v, SerialiseValue v) => Proxy v -> [RawBytes] -> Bool
 serialiseValueConcatDistributes _ xs = deserialiseValueN @v xs == deserialiseValue (mconcat xs)
 
+{-------------------------------------------------------------------------------
+  RawBytes
+-------------------------------------------------------------------------------}
+
+-- | @'packSlice' prefix x suffix@ makes @x@ into a slice with @prefix@ bytes on
+-- the left and @suffix@ bytes on the right.
+packSlice :: RawBytes -> RawBytes -> RawBytes -> RawBytes
+packSlice prefix x suffix =
+    RB.take (RB.size x) (RB.drop (RB.size prefix) (prefix <> x <> suffix))
+
+{-------------------------------------------------------------------------------
+  Error messages
+-------------------------------------------------------------------------------}
+
+inputBytesNotPrecisely :: String -> String -> Int -> Int -> String
+inputBytesNotPrecisely funcName tyName actual expected =
+      showString funcName
+    . showSpace
+    . showParen True (showString tyName)
+    . showString ": expected "
+    . showInt actual
+    . showString " bytes, but got "
+    . showInt expected
+    $ ""
```

Comment:
```suggestion
    . showString ": expected "
    . showInt expected
    . showString " bytes, but got "
    . showInt actual
    $ ""
```

Also, intuitively I would put the `actual` argument last, probably because everything else is more like "config" for the check and then at the end we perform it.